parse/function.go

```diff
@@ -249,7 +286,20 @@ func walk(expr value.Expr, assign bool, f func(value.Expr, bool)) {
 	case *value.Vector:
 	case *value.Matrix:
 	default:
-		fmt.Printf("unknown %T in references\n", e)
+		fmt.Printf("unknown %T in walk\n", e)
 	}
 	f(expr, assign)
 }
+
+func walkVars(expr value.Expr, f func(*value.VarExpr)) {
```

Comment:
Is this too general? funcArg as well. Recursion allows this, right?

op x (a (b (c (d e)))) = a+b+c+d+e

and that should be an error. If it is an error, where does it get caught?

Reply:
Ivy already allows

    (a (b (c (d e)))) = (1 (2 (3 (4 5))))

as an ordinary assignment, so it seemed most consistent to allow here as well.
You could imagine representing a rectangle as ((xmin ymin) (xmax ymax)) and doing

    op area ((xmin xmax) (ymin ymax)) = ...

instead of

    op area (min max) = (xmin xmax) (ymin ymax) = min max; ...

There is a test case

    op (a (b c)) f ((d e) g) = +/a b c d e g

On the other hand, I haven't needed this general form yet in a program I've written.

Reply:
Wow, ok.